stats/stats.go

```diff
@@ -203,24 +205,61 @@ func NewHistogram(Offset float64, Divider float64) *Histogram {
 	return h
 }
 
-// Tradeoff memory for speed (though that also kills the cache so...)
-// this creates an array of 100k (max value) entries
-// TODO: consider using an interval search for the last N big buckets
+// Val2Bucket values are kept in two different structure
+// val2Bucket allows you reach between 0 and 1000 in constant time
+// linearTimeVal2Bucket allows you to reach between 1000 and 100000 in linear time
+// linearTimeVal2Bucket approach saves memory usage but loses speed
+// TODO: change linear time to O(log(N)) with binary search for linearTimeVal2Bucket
 func init() {
-	lastV := int32(lastValue)
-	val2Bucket = make([]int, lastV+1)
+	maxArrayValueIndex := 0
```

Comment:
initialize to -1 (because 0 is valid and you want to catch not found fast)

Reply:
May I keep maxArrayValueIndex instead of maxArrayValue as constant? because when I use lookup, I will always need to look for maxArrayValue index.

Reply:
Nevermind, the values are not constant. I can set when ever I want. I thought they were constant :).